### ISSUE TITLE
run-scaled no longer works

### ISSUE THREAD

**Issue by expipiplus1** [NixOS/nixpkgs#122159]
**To Reproduce**

- Run: `nix-shell -p run-scaled --run run_scaled gnome-calculator`
- Observe

```
using systemd-run to wrap 'start' server command
'systemd-run' '--description' 'xpra-start' '--scope' '--user' 'python3.8' '/nix/store/f6xz7a877cljh2ij1yhrfkd3flsqzp2i-xpra-4.0.6/bin/xpra' 'start' ':2448249' '--xvfb=Xvfb +extension Composite -screen 0 1920x1080x24+32 -nolisten tcp -noreset  -auth $XAUTHORITY' '--env=GDK_SCALE=1' '--env=GDK_DPI_SCALE=1' '--start-child=gnome-calculator ' '--exit-with-children=yes' '--systemd-run=no'
Running scope as unit: run-r2bae6fac87b544b28c93d543e1edbab1.scope
  File "/nix/store/f6xz7a877cljh2ij1yhrfkd3flsqzp2i-xpra-4.0.6/bin/xpra", line 2
    export PATH='/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/bin:/nix/store/f6xz7a877cljh2ij1yhrfkd3flsqzp2i-xpra-4.0.6/bin:/nix/store/cx0py33vl5li9yinz67rbky8g0lrv3wf-python3.8-numpy-1.20.2/bin:/nix/store/0dj85avpdlzrh97fmp7sh02a0lz1z5nv-cairo-1.16.0-dev/bin:/nix/store/bpzgxz0g1150vzw5nv2xzvch679dx256-freetype-2.10.4-dev/bin:/nix/store/p5fz1xbax22kas6dbpbds3k47yl3dd4b-bzip2-1.0.6.0.2-bin/bin:/nix/store/ysjkgs78f3p1c7fpr3bncqbfsy3afanc-libpng-apng-1.6.37-dev/bin:/nix/store/h7914crdcalh0naq2warphlclnyvmpa4-fontconfig-2.13.92-bin/bin:/nix/store/h8m1lraq1lrxrxyc815nxbgqj6m5jrmr-expat-2.2.10-dev/bin:/nix/store/wkw8b59nljjv649vwf2v05i8qkx5p1ns-glib-2.66.8-dev/bin:/nix/store/wzfdkw1ia9x7jdrvxb5jp5wx3xl3ia1x-gettext-0.21/bin:/nix/store/q2fpqg99gfn0asbsjqb2f03vzs4n7cz4-glib-2.66.8-bin/bin:/nix/store/x4y3ilqxixdmqabah2y3mv0zah58bsln-gstreamer-1.18.2-dev/bin:/nix/store/rhmhij476gmycnx7ra1s86cydz2mi3ig-gst-plugins-base-1.18.2/bin:/nix/store/90af71mbpn7h8b8fjd5yfsm29dhcvbl6-linux-pam-1.5.1/bin:/nix/store/pl7q78kz0yj95s1721s3d5jh9kvmpz0l-opencv-4.5.2/bin'${PATH:+':'}$PATH
           ^
SyntaxError: invalid syntax
2021-05-08 14:56:12,638 Error: cannot import X11 bindings:
2021-05-08 14:56:12,638  /nix/store/f6xz7a877cljh2ij1yhrfkd3flsqzp2i-xpra-4.0.6/lib/python3.8/site-packages/xpra/x11/gtk3/gdk_bindings.cpython-38-x86_64-linux-gnu.so: undefined symbol: XDamageQueryExtension
2021-05-08 14:56:12,675 Xpra GTK3 X11 client version 4.0.6-r28285 64-bit
2021-05-08 14:56:12,694  running on Linux unknown unknown unknown
2021-05-08 14:56:12,696  window manager is 'GNOME Shell'
2021-05-08 14:56:13,165 GStreamer version 1.18.2 for Python 3.8.9 64-bit
2021-05-08 14:56:13,192 Warning: failed to load the GTK3 notification class
2021-05-08 14:56:13,192  Namespace Notify not available
2021-05-08 14:56:13,310 No OpenGL_accelerate module loaded: No module named 'OpenGL_accelerate'
2021-05-08 14:56:13,476 Warning: missing gdk bindings:
2021-05-08 14:56:13,476  /nix/store/f6xz7a877cljh2ij1yhrfkd3flsqzp2i-xpra-4.0.6/lib/python3.8/site-packages/xpra/x11/gtk3/gdk_bindings.cpython-38-x86_64-linux-gnu.so: undefined symbol: XDamageQueryExtension
2021-05-08 14:56:13,505 OpenGL enabled with GeForce RTX 2080 Ti/PCIe/SSE2
2021-05-08 14:56:13,560 Error: cannot watch for video device changes without pyinotify:
2021-05-08 14:56:13,560  No module named 'pyinotify'
2021-05-08 14:56:13,579 server socket for display :2448249 not found
2021-05-08 14:56:13,579  waiting up to 20 seconds
2021-05-08 14:56:33,647 Warning: failed to connect:
2021-05-08 14:56:33,647  cannot find live server for display :2448249
server socket for display :2448249 not found
 waiting up to 20 seconds
Warning: failed to connect:
 cannot find live server for display :2448249
```

**Expected behavior**

gnome calculator starts

**Notify maintainers**

run-scaled: @snaar 
xpra: @tstrobel @offline @numinit

**Metadata**

```
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.30, NixOS, 21.05pre287585.edb9e32ff1e (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-21.05pre287333.63586475587"`
 - channels(j): `""`
 - nixpkgs: `/home/j/src/nixpkgs`

```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- run_scaled
# a list of nixos modules affected by the problem
module:
```

**Comment by expipiplus1** [comment]
Seems to be an `xpra` issue. The first (of several) issue is that it tries to call itself as a python module, but we've wrapped it!

This can be cleverly worked around by making the `xpra` executable a bash/python polyglot!

```
diff --git a/pkgs/tools/X11/xpra/default.nix b/pkgs/tools/X11/xpra/default.nix
index 0dffdf7bbf7..d74b5a0c7bf 100644
--- a/pkgs/tools/X11/xpra/default.nix
+++ b/pkgs/tools/X11/xpra/default.nix
@@ -4,7 +4,8 @@
 , ffmpeg, x264, libvpx, libwebp, x265
 , libfakeXinerama
 , gst_all_1, pulseaudio, gobject-introspection
-, pam }:
+, pam
+, bash }:

 with lib;

@@ -100,6 +101,16 @@ in buildPythonApplication rec {
       --prefix PATH : ${lib.makeBinPath [ getopt xorgserver xauth which util-linux pulseaudio ]}
     )
   '';
+  postFixup = ''
+    mv $out/bin/xpra $out/bin/xpra.tmp
+    echo "#!${bash}/bin/bash" > $out/bin/xpra
+    echo "'''''':" >> $out/bin/xpra
+    cat $out/bin/xpra.tmp >> $out/bin/xpra
+    echo "exit\n''''" >> $out/bin/xpra
+    cat $out/bin/.xpra-wrapped >> $out/bin/xpra
+    rm $out/bin/xpra.tmp
+    chmod +x $out/bin/xpra
+  '';

   # append module paths to xorg.conf
   postInstall = ''
```

**Comment by snaar** [comment]
Hi! I assume this is for xpra maintainers first? Or is there preference to work around it in the run-scaled instead?

**Comment by expipiplus1** [comment]
Yeah, I don't think this is actually anything specific to run scaled.

**Comment by milahu** [comment]
still an issue. simply running `xpra start` throws the `SyntaxError`

> making the `xpra` executable a bash/python polyglot

nice : )

```nix
{
  xpraPatched = xpra.overridePythonAttrs (old: {
    # https://github.com/NixOS/nixpkgs/issues/122159
    postFixup = ''
      cat >$out/bin/.xpra-polyglot <<EOF
      $(head -n1 $out/bin/xpra)
      # polyglot: bash + python
      '''':
      # bash
      $(tail -n +2 $out/bin/xpra)
      exit
      '''
      # python
      $(cat $out/bin/.xpra-wrapped)
      EOF
      mv $out/bin/.xpra-polyglot $out/bin/xpra
      chmod +x $out/bin/xpra
    '';
  });
```

**Comment by snaar** [comment]
Seems to be fixed in #172682 - certainly works in 22.05
